source/simulator/solver.cc

```diff
@@ -324,13 +331,29 @@ namespace aspect
         {
           SolverControl solver_control(10000, utmp.l2_norm() * A_block_tolerance);
           PrimitiveVectorMemory<LinearAlgebra::Vector> mem;
-          SolverCG<LinearAlgebra::Vector> solver(solver_control,mem);
 
           try
             {
               dst.block(0) = 0.0;
-              solver.solve(stokes_matrix.block(0,0), dst.block(0), utmp,
-                           a_preconditioner);
+
+              if (A_block_is_symmetric)
+                {
+                  SolverCG<LinearAlgebra::Vector> solver(solver_control,mem);
+                  solver.solve(stokes_matrix.block(0,0), dst.block(0), utmp,
+                               a_preconditioner);
+                }
+              else
+                {
+                  // Use BiCGStab for non-symmetric matrices.
+                  // Do not compute the exact residual, as this
+                  // is more expensive, and we only need an approximate solution.
+                  SolverBicgstab<LinearAlgebra::Vector>
+                  solver(solver_control,
+                         mem,
+                         SolverBicgstab<LinearAlgebra::Vector>::AdditionalData(/*exact_residual=*/ false));
+                  solver.solve(stokes_matrix.block(0,0), dst.block(0), utmp,
+                               a_preconditioner);
```

Comment:
I imagine `a_preconditioner` here is the Trilinos AMG? If so, are we building that with a flag that says that the matrix is symmetric?

Reply:
I couldnt find such a flag for the deal.II `PreconditionAMG` class, and neither could I find one in ASPECT. What would that flag be? The flags we set are in source/simulator/assembly.cc:444-510.

Reply:
Oh, ok. I remembered that one of them did.